allennlp/common/file_utils.py

```diff
@@ -233,9 +235,45 @@ def cached_path(
     cache_dir = os.path.expanduser(cache_dir)
     os.makedirs(cache_dir, exist_ok=True)
 
+    extraction_path: Optional[str] = None
+
     if not isinstance(url_or_filename, str):
         url_or_filename = str(url_or_filename)
 
+    if not os.path.isfile(url_or_filename) and not urlparse(url_or_filename).scheme in ("http", "https", "s3"):
```

Comment:
I am a little worried about this. It's not transparent to a user that the behavior of this hinges on the presence of absence of a local file in their current working directory. It also makes the AllenNLP configuration files a lot less self-contained.

Do you think we could treat this like a URL, and make it `huggingface://lysandre/pair-classification-roberta-mnli` or `hf://lysandre/pair-classification-roberta-mnli` or even `hub://lysandre/pair-classification-roberta-mnli`?

Reply:
Yes, we could definitely treat it like an URL and have it preceded by either of the three propositions you have made. Is there one you favor?

Reply:
I think `hub` is too generic. So I vote for either `huggingface://`, `hf://`, or both.